src/tests/end-to-end/common/browser-factory.ts

```diff
@@ -1,32 +1,54 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
+import { generateUID } from 'common/uid-generator';
 import * as fs from 'fs';
 import * as path from 'path';
 import * as Puppeteer from 'puppeteer';
 import * as util from 'util';
-import { generateUID } from '../../../common/uid-generator';
 import { Browser } from './browser';
 import { popupPageElementIdentifiers } from './element-identifiers/popup-page-element-identifiers';
 import { DEFAULT_BROWSER_LAUNCH_TIMEOUT_MS } from './timeouts';
 
 export const chromeLogsPath = path.join(__dirname, '../../../../test-results/e2e/chrome-logs/');
 
-export function browserLogPath(browserInstanceId: string): string {
-    return path.join(chromeLogsPath, browserInstanceId);
-}
+export const browserLogPath = (browserInstanceId: string): string => path.join(chromeLogsPath, browserInstanceId);
+
+const fileExists = util.promisify(fs.exists);
+const writeFile = util.promisify(fs.writeFile);
+const readFile = util.promisify(fs.readFile);
 
 export interface ExtensionOptions {
     suppressFirstTimeDialog: boolean;
+    addLocalhostToPermissions?: boolean;
```

Comment:
I think it would probably help prevent bugs for this to be non-optional; if a test wants to not have the extra permissions, it has to choose that explicitly

Reply:
I see it the other way around: we don't give those permissions, unless the test explicitly set this flag to `true`

Reply:
I don't think it should default to true, I'm saying I think it should maybe not have any default value (force every test to choose explicitly)

Reply:
It defaults to `false` and having optional make test that don't need this permissions less verbose about it.